src/transaction.c

```diff
@@ -552,10 +1348,11 @@ int wally_tx_remove_input(struct wally_tx *tx, size_t index)
 int wally_tx_add_output(struct wally_tx *tx, const struct wally_tx_output *output)
 {
     uint64_t total;
-
-    if (!is_valid_tx(tx) || !is_valid_tx_output(output) ||
-        wally_tx_get_total_output_satoshi(tx, &total) != WALLY_OK ||
-        total + output->satoshi < total || total + output->satoshi > WALLY_SATOSHI_MAX)
+    const bool is_elements = output->features & WALLY_TX_IS_ELEMENTS;
+    if (!is_valid_tx(tx) || (!is_valid_tx_output(output) && !is_valid_elements_tx_output(output)) ||
+        ((wally_tx_get_total_output_satoshi(tx, &total) != WALLY_OK ||
+          (total + output->satoshi < total || total + output->satoshi > WALLY_SATOSHI_MAX)) &&
```

Comment:
Was it intended to exclude elements from the overflow check as well as the > WALLY_SATOSHI_MAX check?